rules/PHPUnit90/Rector/MethodCall/ReplaceAtMethodWithDesiredMatcherRector.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\PHPUnit\PHPUnit90\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Scalar\Int_;
+use Rector\PHPUnit\NodeAnalyzer\TestsNodeAnalyzer;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\PHPUnit\Tests\PHPUnit90\Rector\MethodCall\ReplaceAtMethodWithDesiredMatcherRector\ReplaceAtMethodWithDesiredMatcherRectorTest
+ */
+final class ReplaceAtMethodWithDesiredMatcherRector extends AbstractRector
+{
+    public function __construct(
+        private readonly TestsNodeAnalyzer $testsNodeAnalyzer
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace at() method call with desired matcher',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+$mock->expects($this->at(0))
+    ->method('foo')
+    ->willReturn('1');
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$mock->expects($this->never())
+    ->method('foo')
+    ->willReturn('1');
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class];
+    }
+
+    /**
+     * @param  MethodCall  $node
+     */
+    public function refactor(Node $node): null|MethodCall
+    {
+        if (! $this->testsNodeAnalyzer->isInTestClass($node)) {
+            return null;
+        }
+
+        if ($node->var instanceof MethodCall && $arg = $this->findAtMethodCall($node->var)) {
+            $this->replaceWithDesiredMatcher($arg);
```

Comment:
return `$node` is needed after node changed, the flag like $hasChanged is needed for it.

Reply:
I created PR :

- https://github.com/rectorphp/rector-phpunit/pull/455

for it.